File: contest/abc280/b.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	in := bufio.NewReader(os.Stdin)
	out := bufio.NewWriter(os.Stdout)
	defer out.Flush()

	var n int
	fmt.Fscan(in, &n)

	before := 0
	arr := []string{}
	for i := 0; i < n; i++ {
		var nn int
		fmt.Fscan(in, &nn)

		arr = append(arr, strconv.Itoa(nn-before))
		before = nn
	}

	fmt.Fprintln(out, strings.Join(arr, " "))
}

File: contest/abc280/b.rb
# n = gets.chomp.to_i
# arr = gets.chomp.split(" ").map(&:to_i)

n = gets.chomp.to_i
arr = gets.chomp.split(" ").map(&:to_i)

before = 0
result = []
arr.each do |a|
  result << a - before
  before = a
end

puts result.join(" ")
